django/contrib/auth/middleware.py

```diff
@@ -34,6 +38,75 @@ def process_request(self, request):
         request.auser = partial(auser, request)
 
 
+class LoginRequiredMiddleware(MiddlewareMixin):
+    """
+    Middleware that forces all views to require login by default.
+
+    Views that have login_not_required decorator will allow unauthenticated
+    requests. Otherwise, it will redirect to a login page.
+    """
+
+    login_url = None
+    permission_denied_message = ""
+    raise_exception = False
+    redirect_field_name = REDIRECT_FIELD_NAME
+
+    def process_view(self, request, view_func, view_args, view_kwargs):
+        if request.user.is_authenticated:
+            return None
+
+        if not getattr(view_func, "login_required", True):
+            return None
+
+        return self.handle_no_permission(request, view_func)
+
+    def get_login_url(self, view_func):
+        login_url = getattr(view_func, "login_url", None)
+        if hasattr(view_func, "view_class"):
+            login_url = getattr(view_func.view_class, "login_url", login_url)
+        login_url = login_url or self.login_url or settings.LOGIN_URL
+        if not login_url:
+            raise ImproperlyConfigured(
+                f"{self.__class__.__name__} is missing the login_url attribute. Define "
+                f"{self.__class__.__name__}.login_url, settings.LOGIN_URL, or override "
+                f"{self.__class__.__name__}.get_login_url()."
+            )
+        return str(login_url)
+
+    def get_redirect_field_name(self, view_func):
+        """
+        Override this method to override the redirect_field_name attribute.
+        """
+        redirect_field_name = getattr(view_func, "redirect_field_name", None)
+        view_class = getattr(view_func, "view_class", None)
+        if view_class:
+            redirect_field_name = getattr(
+                view_class, "redirect_field_name", redirect_field_name
+            )
+        return redirect_field_name or self.redirect_field_name
```

Comment:
Ditto.

```suggestion
        return (
            getattr(view_func, "redirect_field_name", None)
            or self.redirect_field_name
        )
```

Reply:
Maybe I'm shaving microseconds for no reason but could you then use `return getattr(view_func, "redirect_field_name", self.redirect_field_name)`?

Reply:
I tried to change this but `redirect_field_name` can be an attribute that is `None` which then makes this return `None`